gh-pages-stub/public/js/search.js

```diff
@@ -0,0 +1,93 @@
+(function() {
+
+  function isBlank(str) {
+    return (!str || /^\s*$/.test(str));
+  } 
+
+  function displaySearchResults(results, store, searchTerm) {
+    var searchResults = document.getElementById('search-results');
+
+    if (results.length) {
+      var appendString = '';
+
+      for (var i = 0; i < results.length; i++) {
+        var item = store[results[i].ref];
+        if (isBlank(item.title)) {
+            item.title = "No title";
+        }
+        appendString += '<li><a href="' + item.url + '"><h3>' + item.title + '</h3></a>';
+	      var indx = item.content.indexOf(searchTerm);
+	      var start_pos = Math.max(0, indx - 75);
+	      var stop_pos = Math.min(item.content.length, indx + 75);
+	      var str_to_add = item.content.substring(start_pos, stop_pos);
+	      str_to_add = str_to_add.replace(searchTerm, "<b><u>" + searchTerm + "</u></b>" );
+        appendString += '<p>';
+        if (start_pos > 0 ) {
+            appendString += '... ';
+	      }
+        appendString += str_to_add;
+        if (stop_pos < item.content.length) {
+           appendString += ' ...'  
+        }
+	      appendString += '</p></li>';
+      }
+      searchResults.innerHTML = appendString;
+    } else {
+      searchResults.innerHTML = '<li>No results found</li>';
+    }
+  }
+
+  function getQueryVariable(variable) {
+    var query = window.location.search.substring(1);
+    var vars = query.split('&');
+
+    for (var i = 0; i < vars.length; i++) {
```

Comment:
you can use map and filter here, instead of for, "for" better to use in places where we have more then 10000 items